modules/core/src/main/java/org/apache/synapse/ExtendedSynapseHandler.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *   * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.synapse;
+
+/**
+ * Interface For Synapse Handlers
+ * <p/>
+ * Synapse Handlers are invoked when a message received to the mediation engine or a message sent out
+ * from the engine.
+ * <p/>
+ * When a message received to the engine, handles are invoked just before the mediation flow
+ * When a message is sent out from the engine, handlers are invoked just after the mediation flow
+ * There are two requests coming into the engine and two responses going out from the engine.
+ * For all four messages relevant method is invoked
+ */
+public interface ExtendedSynapseHandler extends SynapseHandler {
+
+    /**
+     * Handle response message going out from the engine
+     *
+     * @param synCtx outgoing response message context
+     * @return whether mediation flow should continue
+     */
+    public boolean handleError(MessageContext synCtx);
+
+    /**
+     * Handle server startup
+     *
+     */
+    public boolean init();
+
+    /**
+     * Handle artifact deployment
+     *
+     * @param artifactName name of the artifact deployed
+     * @param artifactType type of the artifact deployed
+     * @param startTime artifact deployed time
+     * @return whether mediation flow should continue
+     */
+    public boolean deployArtifacts(String artifactName, String artifactType, String startTime);
```

Comment:
What is the use case of this method ?

Reply:
We call this method when a service is deployed to get the information required for generating Grafana Dashboards.

Reply:
.How do we handler redeployment and un deployments , doesn't they matter. Also could you add these information to the PR description/

Reply:
When an undeployment happens when scraping Prometheus detects it as no data. So it's not scraped.